x/upgrade/depinject.go

```diff
@@ -57,7 +59,13 @@ func ProvideModule(in ModuleInputs) ModuleOutputs {
 		skipUpgradeHeights = make(map[int64]bool)
 	)
 
-	if in.AppOpts != nil {
+	if in.Viper != nil { // viper takes precedence over app options
+		for _, h := range in.Viper.GetIntSlice(server.FlagUnsafeSkipUpgrades) {
```

Comment:
should we make the flags in `x/group` so don't need to care about v1 or v2 dep in the future?

Reply:
(Assuming you meant x/upgrade), What do you mean?